Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -203,47 +203,46 @@ function xASL_wrp_ResampleASL(x)
 
 
 %% ------------------------------------------------------------------------------------------
-% 8. Pair-wise subtraction & saving PWI & PWI4D in both spaces
+% 8. Pair-wise control-label subtraction
+% for both native & standard space
+
+% Nomenclature:
+% ASL4D = original timeseries
+% PWI4D = subtracted timeseries -> for quantification purposes
+% PWI3D = averaged per TE—PLD—LabDur combination -> for QC purposes
+% PWI   = single perfusion-weighted volume -> for registration purposes
+
+% 
+
 % Load ASL time series (after being pre-processed)
 StringSpaceIs = {'native' 'standard'};
-PathASL = {x.P.Path_rdespiked_ASL4D x.P.Path_rtemp_despiked_ASL4D};
+PathASL4D = {x.P.Path_rdespiked_ASL4D x.P.Path_rtemp_despiked_ASL4D};
 PathPWI = {x.P.Path_PWI x.P.Pop_Path_PWI};
+PathPWI3D = {x.P.Path_PWI3D x.P.Pop_Path_PWI3D};
 PathPWI4D = {x.P.Path_PWI4D x.P.Pop_Path_PWI4D};
 
 for iSpace=1:2
     fprintf('%s\n', ['Saving in ' StringSpaceIs{iSpace} ' space:']);
     
-    ASL_im = xASL_io_Nifti2Im(PathASL{iSpace}); % Load time-series nifti
+    ASL_im = xASL_io_Nifti2Im(PathASL4D{iSpace}); % Load time-series nifti
     dim4 = size(ASL_im, 4);
 
-    if dim4==1
-        % Apparently, the subtraction was already done on the scanner/reconstruction
-        fprintf('%s\n', PathPWI{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, ASL_im, 32, false);
-	elseif x.modules.asl.bContainsDeltaM

```

Comment:
Why have you deleted this part? We need this for sequences with deltaM. WIthout this option, you will produce an unnecessary warning.

Reply:
Now I see that you have moved this below, but that's not really absolutely correct as you can have more than 1 deltaM.

Reply:
Fixed in e9bf37aa3f3ef7837a2f0304a17d0d2f1b20342c by only skipping the subtraction if bContainsDeltaM

Reply:
Perhaps you should use the modulo instead ;)

Reply:
Added this to the #997 issue requirement list

Reply:
OK. But please still keep this issue open.

Reply:
This should be fixed now. if bContainsDeltaM, the subtraction is skipped.

Reply:
A few errors and special cases had to be fixed for data containing deltaM or CBF.